lib/screens/home/home_bloc.dart

```diff
@@ -19,21 +21,27 @@ class HomeBloc extends Bloc<HomeEvent, HomeState> {
           sortBy: homePreferences.sortType,
           filterBy: homePreferences.filterType));
     });
+    _pageRepository = PageRepository(client: client);
   }
 
   @override
-  Stream<HomeState> mapEventToState(HomeEvent event,) async* {
+  Stream<HomeState> mapEventToState(HomeEvent event) async* {
     if (event is GetPageOfPackagesEvent) {
-      Page page = await client.getPageOfPackages(
-        pageNumber: event.pageNumber,
-        sortBy: event.sortBy,
-        filterBy: event.filterBy,
-      );
+      Page page;
```

Comment:
since this is a `HydratedBloc` you can now only update if the currentState doesn't have data

```dart
if (currentState.page != null) return; // don't re-fetch if we already have a cached page.
```

Reply:
I don't understand. Update what?